### PR TITLE
Validate IO after resource deletion

### PR DESCRIPTION
**Pull request by ebenahar**

### CODE REVIEW

**Comment by ebenahar** [comment]
```
(ocs-venv) [ebenahar@localhost ocs-ci]$ run-ci --cluster-name ebenahar-ocs-deployment --cluster-path ~/ebenahar-ocs-dir/ tests/manage/test_pvc_disruptive.py::TestRBDDisruption::test_disruptive_block[run_io-mon] --no-print-logs --capture=no           
Collecting Cluster versions
=============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.7.2, pytest-4.6.3, py-1.8.0, pluggy-0.12.0
rootdir: /home/ebenahar/git/ocs-ci, inifile: pytest.ini
plugins: html-1.21.1, logger-0.5.0, reportportal-1.0.5, metadata-1.8.0
collected 1 item                                                                                                                                                                                                  

tests/manage/test_pvc_disruptive.py::TestRBDDisruption::test_disruptive_block[run_io-mon] 
------------------------------------------------------------------------------------------------- live log setup --------------------------------------------------------------------------------------------------
21:37:59 - MainThread - tests.conftest - INFO - All logs located at /tmp/logs_1562783865
21:37:59 - MainThread - tests.conftest - INFO - Running OCS basic installation
21:37:59 - MainThread - ocs_ci.ocs.openshift_ops - INFO - Testing access to cluster with /home/ebenahar/ebenahar-ocs-dir/auth/kubeconfig
21:37:59 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc cluster-info
21:38:00 - MainThread - ocs_ci.ocs.openshift_ops - INFO - Access to cluster is OK!
21:38:00 - MainThread - tests.conftest - INFO - The installation is skipped because the cluster is running
21:38:00 - Dummy-1 - ocs_ci.utility.utils - INFO - Executing command: oc --kubeconfig /home/ebenahar/ebenahar-ocs-dir/auth/kubeconfig get Pod -A -o yaml
21:38:00 - Dummy-2 - ocs_ci.utility.utils - INFO - Executing command: oc --kubeconfig /home/ebenahar/ebenahar-ocs-dir/auth/kubeconfig get StorageClass -A -o yaml
21:38:00 - Dummy-3 - ocs_ci.utility.utils - INFO - Executing command: oc --kubeconfig /home/ebenahar/ebenahar-ocs-dir/auth/kubeconfig get CephFileSystem -A -o yaml
21:38:00 - Dummy-4 - ocs_ci.utility.utils - INFO - Executing command: oc --kubeconfig /home/ebenahar/ebenahar-ocs-dir/auth/kubeconfig get CephBlockPool -A -o yaml
21:38:00 - Dummy-5 - ocs_ci.utility.utils - INFO - Executing command: oc --kubeconfig /home/ebenahar/ebenahar-ocs-dir/auth/kubeconfig get PersistentVolume -A -o yaml
21:38:00 - Dummy-6 - ocs_ci.utility.utils - INFO - Executing command: oc --kubeconfig /home/ebenahar/ebenahar-ocs-dir/auth/kubeconfig get PersistentVolumeClaim -A -o yaml
21:38:00 - Dummy-7 - ocs_ci.utility.utils - INFO - Executing command: oc --kubeconfig /home/ebenahar/ebenahar-ocs-dir/auth/kubeconfig get Secret -A -o yaml
21:38:00 - Dummy-8 - ocs_ci.utility.utils - INFO - Executing command: oc --kubeconfig /home/ebenahar/ebenahar-ocs-dir/auth/kubeconfig get Namespace -A -o yaml
21:39:22 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/ebenahar/ebenahar-ocs-dir/auth/kubeconfig get Pod --selector=app=rook-ceph-tools -o yaml
21:39:23 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/ebenahar/ebenahar-ocs-dir/auth/kubeconfig rsh rook-ceph-tools-76c7d559b6-6zmpg ceph auth get-key client.admin --format json-pretty
21:39:26 - MainThread - ocs_ci.ocs.resources.ocs - INFO - Adding Secret with name secret-test-rbd-1021392682
21:39:26 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/ebenahar/ebenahar-ocs-dir/auth/kubeconfig create -f /tmp/Secretzjg3evu3 -o yaml
21:39:27 - MainThread - ocs_ci.ocs.resources.ocs - INFO - Adding CephBlockPool with name cbp-test-1021392787
21:39:27 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/ebenahar/ebenahar-ocs-dir/auth/kubeconfig create -f /tmp/CephBlockPoolhdh1pflm -o yaml
21:39:28 - MainThread - tests.helpers - INFO - Verifying that block pool cbp-test-1021392787 exists
21:39:28 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/ebenahar/ebenahar-ocs-dir/auth/kubeconfig get Pod --selector=app=rook-ceph-tools -o yaml
21:39:29 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/ebenahar/ebenahar-ocs-dir/auth/kubeconfig rsh rook-ceph-tools-76c7d559b6-6zmpg ceph osd lspools --format json-pretty
21:39:33 - MainThread - tests.helpers - INFO - POOLS are [{'poolnum': 1, 'poolname': 'rbd'}, {'poolnum': 2, 'poolname': 'ocsci-cephfs-metadata'}, {'poolnum': 3, 'poolname': 'ocsci-cephfs-data0'}, {'poolnum': 5, 'poolname': 'cbp-test-1019274107'}, {'poolnum': 7, 'poolname': 'cbp-test-1021392787'}]
21:39:33 - MainThread - ocs_ci.ocs.resources.ocs - INFO - Adding StorageClass with name storageclass-test-rbd-1021393305
21:39:33 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/ebenahar/ebenahar-ocs-dir/auth/kubeconfig create -f /tmp/StorageClassg08p9a4k -o yaml
-------------------------------------------------------------------------------------------------- live log call --------------------------------------------------------------------------------------------------
21:39:34 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/ebenahar/ebenahar-ocs-dir/auth/kubeconfig get Pod --selector=app=rook-ceph-mon -o yaml
21:39:35 - MainThread - ocs_ci.ocs.resources.ocs - INFO - Adding PersistentVolumeClaim with name pvc-test-1021393528
21:39:35 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/ebenahar/ebenahar-ocs-dir/auth/kubeconfig create -f /tmp/PersistentVolumeClaimbx404emw -o yaml
21:39:36 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/ebenahar/ebenahar-ocs-dir/auth/kubeconfig get PersistentVolumeClaim pvc-test-1021393528 -o yaml
21:39:37 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/ebenahar/ebenahar-ocs-dir/auth/kubeconfig get PersistentVolumeClaim pvc-test-1021393528 -o yaml
21:39:41 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/ebenahar/ebenahar-ocs-dir/auth/kubeconfig get PersistentVolumeClaim pvc-test-1021393528 -o yaml
21:39:42 - MainThread - ocs_ci.ocs.resources.ocs - INFO - Adding Pod with name pod-test-rbd-1021394217
21:39:42 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/ebenahar/ebenahar-ocs-dir/auth/kubeconfig create -f /tmp/POD_s_c4mdhz -o yaml
21:39:43 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/ebenahar/ebenahar-ocs-dir/auth/kubeconfig get Pod pod-test-rbd-1021394217 -o yaml
21:39:44 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/ebenahar/ebenahar-ocs-dir/auth/kubeconfig get Pod pod-test-rbd-1021394217 -o yaml
21:39:48 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/ebenahar/ebenahar-ocs-dir/auth/kubeconfig get Pod pod-test-rbd-1021394217 -o yaml
21:39:57 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/ebenahar/ebenahar-ocs-dir/auth/kubeconfig get Pod pod-test-rbd-1021394217 -o yaml
21:39:58 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/ebenahar/ebenahar-ocs-dir/auth/kubeconfig rsh pod-test-rbd-1021394217 which apt-get
21:40:00 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/ebenahar/ebenahar-ocs-dir/auth/kubeconfig rsh pod-test-rbd-1021394217 apt-get update
21:40:06 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/ebenahar/ebenahar-ocs-dir/auth/kubeconfig rsh pod-test-rbd-1021394217 apt-get -y install fio
21:40:14 - MainThread - ocs_ci.utility.utils - WARNING - Command warning:: debconf: delaying package configuration, since apt-utils is not installed
21:40:14 - ThreadPoolExecutor-1_0 - ocs_ci.utility.workloads.fio - INFO - Running cmd: fio --name=fio-rand-readwrite --filename=/var/lib/www/html --readwrite=randrw --bs=4K --direct=1 --numjobs=1 --time_based=1 --runtime=60 --size=1G --iodepth=4 --invalidate=1 --fsync_on_close=1 --rwmixread=75 --ioengine=libaio --output-format=json
21:40:14 - MainThread - ocs_ci.ocs.workload - INFO - Done submitting..
21:40:14 - ThreadPoolExecutor-1_0 - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/ebenahar/ebenahar-ocs-dir/auth/kubeconfig rsh pod-test-rbd-1021394217 fio --name=fio-rand-readwrite --filename=/var/lib/www/html --readwrite=randrw --bs=4K --direct=1 --numjobs=1 --time_based=1 --runtime=60 --size=1G --iodepth=4 --invalidate=1 --fsync_on_close=1 --rwmixread=75 --ioengine=libaio --output-format=json
21:40:14 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/ebenahar/ebenahar-ocs-dir/auth/kubeconfig delete Pod rook-ceph-mon-a-5c496669cb-kgzjj --grace-period=0 --force
21:40:16 - MainThread - ocs_ci.utility.utils - WARNING - Command warning:: warning: Immediate deletion does not wait for confirmation that the running resource has been terminated. The resource may continue to run on the cluster indefinitely.
21:40:16 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/ebenahar/ebenahar-ocs-dir/auth/kubeconfig get Pod --selector=app=rook-ceph-mon -o yaml
21:40:20 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/ebenahar/ebenahar-ocs-dir/auth/kubeconfig get Pod --selector=app=rook-ceph-mon -o yaml
21:40:24 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/ebenahar/ebenahar-ocs-dir/auth/kubeconfig get Pod --selector=app=rook-ceph-mon -o yaml
21:40:28 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/ebenahar/ebenahar-ocs-dir/auth/kubeconfig get Pod --selector=app=rook-ceph-mon -o yaml
21:40:33 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/ebenahar/ebenahar-ocs-dir/auth/kubeconfig get Pod --selector=app=rook-ceph-mon -o yaml
21:40:34 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/ebenahar/ebenahar-ocs-dir/auth/kubeconfig rsh pod-test-rbd-1021394217 which apt-get
21:40:37 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/ebenahar/ebenahar-ocs-dir/auth/kubeconfig rsh pod-test-rbd-1021394217 apt-get update
21:40:41 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/ebenahar/ebenahar-ocs-dir/auth/kubeconfig rsh pod-test-rbd-1021394217 apt-get -y install fio
21:40:44 - ThreadPoolExecutor-2_0 - ocs_ci.utility.workloads.fio - INFO - Running cmd: fio --name=fio-rand-readwrite --filename=/var/lib/www/html --readwrite=randrw --bs=4K --direct=1 --numjobs=1 --time_based=1 --runtime=60 --size=1G --iodepth=4 --invalidate=1 --fsync_on_close=1 --rwmixread=75 --ioengine=libaio --output-format=json
21:40:44 - MainThread - ocs_ci.ocs.workload - INFO - Done submitting..
21:40:44 - ThreadPoolExecutor-2_0 - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/ebenahar/ebenahar-ocs-dir/auth/kubeconfig rsh pod-test-rbd-1021394217 fio --name=fio-rand-readwrite --filename=/var/lib/www/html --readwrite=randrw --bs=4K --direct=1 --numjobs=1 --time_based=1 --runtime=60 --size=1G --iodepth=4 --invalidate=1 --fsync_on_close=1 --rwmixread=75 --ioengine=libaio --output-format=json
21:40:44 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/ebenahar/ebenahar-ocs-dir/auth/kubeconfig delete Pod pod-test-rbd-1021394217
21:40:49 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/ebenahar/ebenahar-ocs-dir/auth/kubeconfig delete PersistentVolumeClaim pvc-test-1021393528
PASSED
------------------------------------------------------------------------------------------------ live log teardown ------------------------------------------------------------------------------------------------
21:40:50 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/ebenahar/ebenahar-ocs-dir/auth/kubeconfig get StorageClass storageclass-test-rbd-1021393305 -o yaml
21:40:51 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/ebenahar/ebenahar-ocs-dir/auth/kubeconfig delete StorageClass storageclass-test-rbd-1021393305
21:40:52 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/ebenahar/ebenahar-ocs-dir/auth/kubeconfig get CephBlockPool cbp-test-1021392787 -o yaml
21:40:53 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/ebenahar/ebenahar-ocs-dir/auth/kubeconfig delete CephBlockPool cbp-test-1021392787
21:40:54 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/ebenahar/ebenahar-ocs-dir/auth/kubeconfig delete Secret secret-test-rbd-1021392682
21:40:56 - Dummy-9 - ocs_ci.utility.utils - INFO - Executing command: oc --kubeconfig /home/ebenahar/ebenahar-ocs-dir/auth/kubeconfig get Pod -A -o yaml
21:40:56 - Dummy-6 - ocs_ci.utility.utils - INFO - Executing command: oc --kubeconfig /home/ebenahar/ebenahar-ocs-dir/auth/kubeconfig get StorageClass -A -o yaml
21:40:56 - Dummy-10 - ocs_ci.utility.utils - INFO - Executing command: oc --kubeconfig /home/ebenahar/ebenahar-ocs-dir/auth/kubeconfig get CephFileSystem -A -o yaml
21:40:56 - Dummy-5 - ocs_ci.utility.utils - INFO - Executing command: oc --kubeconfig /home/ebenahar/ebenahar-ocs-dir/auth/kubeconfig get CephBlockPool -A -o yaml
21:40:56 - Dummy-8 - ocs_ci.utility.utils - INFO - Executing command: oc --kubeconfig /home/ebenahar/ebenahar-ocs-dir/auth/kubeconfig get PersistentVolume -A -o yaml
21:40:56 - Dummy-3 - ocs_ci.utility.utils - INFO - Executing command: oc --kubeconfig /home/ebenahar/ebenahar-ocs-dir/auth/kubeconfig get PersistentVolumeClaim -A -o yaml
21:40:56 - Dummy-1 - ocs_ci.utility.utils - INFO - Executing command: oc --kubeconfig /home/ebenahar/ebenahar-ocs-dir/auth/kubeconfig get Secret -A -o yaml
21:40:56 - Dummy-11 - ocs_ci.utility.utils - INFO - Executing command: oc --kubeconfig /home/ebenahar/ebenahar-ocs-dir/auth/kubeconfig get Namespace -A -o yaml


=========================================================================================== 1 passed in 216.41 seconds ============================================================================================
```

File: tests/manage/test_pvc_disruptive.py [hunk 1]
```diff
@@ -50,6 +50,9 @@ def disruptive_base(self, operation_to_disrupt, resource_to_delete):
         if operation_to_disrupt == 'run_io':
             DISRUPTION_OPS.delete_resource()
 
+            # Validate that it's possible to run IO again
+            self.pod_obj.run_io(storage_type=self.storage_type, size='1G')
```
Comment (RazTamir) [inline tests/manage/test_pvc_disruptive.py]: maybe wait for results?
use short runtime

**Comment by vasukulkarni** [comment]
In the logs, I dont see FIO stats, I am not sure its actually run

File: tests/manage/test_pvc_disruptive.py (hunk 1 above)
Comment (vasukulkarni) [reply tests/manage/test_pvc_disruptive.py]: I just checked the function, I think it changed a bit from my last review, its no longer a blocking call , so we might need that here or as a different function.

Comment (vasukulkarni) [reply tests/manage/test_pvc_disruptive.py]: we need this https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/ocs/resources/pod.py#L95, but for simplicity I am going to club both calls and send PR

Comment (vasukulkarni) [reply tests/manage/test_pvc_disruptive.py]: I will test the #390 and it should solve the issue, other tests I can set the wait=False if that is what they intend

Comment (vasukulkarni) [reply tests/manage/test_pvc_disruptive.py]: my pods are gone, have to destroy/redeploy :(

Comment (jilju) [reply tests/manage/test_pvc_disruptive.py]: I think storage_type should be 'fs' irrespective of rbd or cephfs.

Comment (sidhant-agrawal) [reply tests/manage/test_pvc_disruptive.py]: Yes, I have also noticed we have to give storage_type 'fs'  for running fio on nginx pod.

**Comment by clacroix12** [comment]
> In the logs, I dont see FIO stats, I am not sure its actually run

@ebenahar Any update on vasu's concern here?

**Comment by ebenahar** [comment]
@clacroix12 the implementation here, along with FIO results info will be covered as part of https://github.com/red-hat-storage/ocs-ci/pull/401/files
Closing

**Pull request closed without merging**